programs/devnet-token-faucet/src/lib.rs

```diff
@@ -73,7 +73,7 @@ pub mod devnet_token_faucet {
                     &[mint_authority.bump],
                 ]],
             ),
-            amount,
+            amount.min(u64::MAX / 1_000_000),
```

Comment:
add one test that requests u64::MAX, assert that it gets no more than `u64::MAX / 10^6`

Reply:
will file a task for it